app/controllers/ProofOfAuthorityController.scala

```diff
@@ -147,17 +145,48 @@ class ProofOfAuthorityController @Inject() (
       renderFileVerificationStatus(reference, request.userAnswers.fileUploadState)
     }
 
+  //GET /upload-proof-of-authority/remove
+  final def onRemove(reference: String): Action[AnyContent] =
+    (identify andThen getData andThen requireData).async { implicit request =>
+      request.userAnswers.fileUploadState match {
+        case Some(fus) =>
+          val acceptedFiles =
+            FileUploaded(fileUploads = fus.fileUploads.copy(files = filesInStateAccepted(fus.fileUploads.files)))
+          val sessionState = SessionState(Some(acceptedFiles), Some(request.userAnswers))
+          fileUtils.applyTransition(
+            removeFileUploadBy(reference)(upscanRequest(request.internalId))(upscanInitiateConnector.initiate(_))(_),
```

Comment:
OBSERVATION: Huh. Well, that smells like a rabbit hole of refactoring at a later date.  That took me a long time to dig down into to find out it does a copy of the files uploaded with a filter on it... (and some upscan dark magic if the list is empty)

Reply:
Indeed.  It is a copy (and slight refactor) of existing "remove file" code used on the existing multi-file upload controllers.
I was going to go further and remove the use of that `SessionState` wrapper  as I don't think it's needed.  But decided against it.